lib/sles4sap/console_redirection.pm

```diff
@@ -129,7 +129,7 @@ sub connect_target_to_serial {
     my $redirect_port = get_required_var("QEMUPORT") + 1;
     my $redirect_ip = get_var('QEMU_HOST_IP', '10.0.2.2');
     my $redirect_opts = "-R $redirect_port:$redirect_ip:$redirect_port";
-    my $switch_root_cmd = $args{switch_root} ? 'sudo su -' : '';
+    my $switch_root_cmd = $args{switch_root} && $args{ssh_user} ne 'root' ? 'sudo su -' : '';
```

Comment:
This avoids logging as root from root user itself.

Reply:
What about using an "assert approach?. 

Now you implemented it like: User can call the function passing `switch_root`, and even when he/she is doing it in conjunction with `ssh_user` --> **root**. Implemented behavior is to internally silently ignore one of the two arguments.

Why not to 
```
died "No need to use switch_root when ssh_user is root" if $args{switch_root} && $args{ssh_user} eq 'root';
```

Reply:
I find it very unnecessary to kill a test for non critical problem. Imagine developing ENSA2 module and a function kills your test after 3h of deployment. 
Is it your fault forgetting your are already on root?  Yes
I it necessary for the function to punish you in such way? No.
I will put there a `record_info` or `diag` warning message instead.